README.md

```diff
@@ -2,10 +2,39 @@
 
 [![codecov](https://codecov.io/gh/macxred/cashctrl_ledger/branch/main/graph/badge.svg)](https://codecov.io/gh/macxred/cashctrl_ledger)
 
-`cashctrl_ledger` is a Python package that implements the `pyledger.LedgerEngine` interface, enabling seamless integration with the CashCtrl accounting service. With this package, users can perform various accounting operations programmatically, directly from Python.
+`cashctrl_ledger` is a Python package that implements the `pyledger.LedgerEngine`
+interface, enabling seamless integration with the CashCtrl accounting service. With this package,
+users can perform various accounting operations programmatically, directly from Python.
+
+## Core Components
+
+1. **CashCtrlLedger**\
+The class that Implements the `pyledger` interface by connecting to the CashCtrl online accounting software via REST API.
+
+2. **CashCtrlLedgerExtended**\
```

Comment:
I am not convinced by name and description of the new class. In one sentence, why do we need this class? What does it do? What is so cool about it?

Reply:
What do you think about the `CashCtrlLedgerHackish` as a better name?

Reply:
CashCtrlLedgerHackish is a terrible name: just states  that the class is in some way improper, without stating what it actually does, what features it contains, or the reasoning why we created it in the first place.

Reply:
#### Class Name Proposals generated with help of ChatGPT

  - [ ] **CashCtrlLedgerSplitter** - Emphasizes the splitting of transactions within the ledger.
  - [ ] **CashCtrlLedgerNormalizer** - Focuses on ensuring transactions conform to CashCtrl's standards.
  - [ ] **CashCtrlLedgerBalancer** - Highlights the balancing of ledger entries after splitting or adjustments.
  - [ ] **CashCtrlTransactionSplitter** - Indicates that the class specifically handles the splitting of transactions.
  - [ ] **CashCtrlTransactionBalancer** - Suggests a focus on balancing transactions, possibly after splitting.
  - [ ] **CashCtrlNormalizeSplitter** - Combines normalization and splitting, indicating both actions are central to the class's functionality.

  **All of these options are valid and align well with the class's intended purpose.**

  You may want to choose based on which aspect of the class’s functionality you want to emphasize most:

  - **Splitting:** `CashCtrlLedgerSplitter`, `CashCtrlTransactionSplitter`
  - **Normalizing:** `CashCtrlLedgerNormalizer`, `CashCtrlNormalizeSplitter`
  - **Balancing:** `CashCtrlLedgerBalancer`, `CashCtrlTransactionBalancer`

Reply:
Also:
- CashCtrlCompliantLedger: This name emphasizes that the ledger is fully compliant with CashCtrl’s rules and restrictions. It suggests that the class ensures all transactions adhere to CashCtrl’s standards.
- EnhancedCashCtrlLedger: This name suggests that the class not only ensures compliance but also extends or improves the base functionality of the CashCtrlLedger. It indicates that this version offers something more than the standard ledger.